### ISSUE TITLE
Missing the code of data train_test_split

### ISSUE THREAD

**Issue by litxiaoyao** [laolintou/scPEFT#1]
It seems that data are all preprocessing before the tutorials notebooks can be used, could you please share the splitted data or the preprocessing code for reproduce the result?
![image](https://github.com/laolintou/scPEFT/assets/68361174/04410eeb-c7a1-422e-a3b7-6d6b08a9cc2d)

**Comment by SELECT-FROM** [comment]
> It seems that data are all preprocessing before the tutorials notebooks can be used, could you please share the splitted data or the preprocessing code for reproduce the result? ![image](https://private-user-images.githubusercontent.com/68361174/316885440-04410eeb-c7a1-422e-a3b7-6d6b08a9cc2d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3MjA1NjUsIm5iZiI6MTcxMTcyMDI2NSwicGF0aCI6Ii82ODM2MTE3NC8zMTY4ODU0NDAtMDQ0MTBlZWItYzdhMS00MjJlLWEzYjctNmQ2YjA4YTljYzJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI5VDEzNTEwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1MmQ5ZDdiYTUyM2VlNzQwYTg2YTU5NjkzMDIzYjdmMGZjNmFmY2FjZTJmMmE1NDhmZGFmNTg4NDdjMzdjNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.81JvvjcW2DhuZPlrp8cEfCrgqNkN68lKyP9SzHGSyjY)

Hi,
The ms dataset comes from the scGPT, and you can download the ms dataset from this link(https://drive.google.com/drive/folders/1Qd42YNabzyr2pWt9xoY4cVMTAxsNBt4v?usp=sharing). We randomly divided 10% of the validation set from the training set using sklearn.model_selection.train_test_split. The preprocessing process fully follows the specifications of scGPT cell type annotation.
Best

**Comment by litxiaoyao** [comment]
Sorry to bother, but I find the code is slightly different from scGPT that it only needs the following two files other than three, and the .obs["batch_id"] is also different. Besides, the key adata.obs["Factor Value[inferred cell type - authors labels]"] is also different from adata.obs["celltype"]. So I guess there seems be more pro-processing codes to get the three files, could you please provide them?
` adata = sc.read(data_dir / "c_data.h5ad")
    adata_test = sc.read(data_dir / "filtered_ms_adata.h5ad")`
vs
`   adata = sc.read(data_dir / f"ms_train.h5ad")
    adata_val = sc.read(data_dir / "ms_val.h5ad")
    adata_test = sc.read(data_dir / "ms_test.h5ad")`
![image](https://github.com/laolintou/scPEFT/assets/68361174/c60d749d-80ad-4ba0-8763-c26ea87c7e17)

**Comment by SELECT-FROM** [comment]
> .obs["batch_id"]
Hi,
I'm sorry for any confusion caused by the previous explanation. In the scGPT cell type annotation tutorial, it requires two inputs for the ms dataset. While in our code, we require three inputs for the ms dataset is to ensure the stability of the reproduction results. We randomly divided the validation set into 10% and saved it as a separate h5ad file. Within adata.obs, the key "Factor Value [inferred cell type - author labels]" is entirely equivalent to the key "celltype." We had previously renamed it to unify the cell type label key, with no other alterations made. Regarding adata.obs["batch_id"], its purpose is to act as an identifier for distinguishing between the train set, validation set, and test set. By assigning a unique batch_id to each set, subsequent splitting becomes easier, allowing for clear identification of the different sets. We are currently engaged in a more substantial project and will gradually unveil more detailed code. Stay tuned for updates! Let me know if you need additional details.
Best

**Comment by SELECT-FROM** [comment]
> Sorry to bother, but I find the code is slightly different from scGPT that it only needs the following two files other than three, and the .obs["batch_id"] is also different. Besides, the key adata.obs["Factor Value[inferred cell type - authors labels]"] is also different from adata.obs["celltype"]. So I guess there seems be more pro-processing codes to get the three files, could you please provide them? ` adata = sc.read(data_dir / "c_data.h5ad") adata_test = sc.read(data_dir / "filtered_ms_adata.h5ad")` vs ` adata = sc.read(data_dir / f"ms_train.h5ad") adata_val = sc.read(data_dir / "ms_val.h5ad") adata_test = sc.read(data_dir / "ms_test.h5ad")` ![image](https://private-user-images.githubusercontent.com/68361174/321175103-c60d749d-80ad-4ba0-8763-c26ea87c7e17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3NjQ5NjIsIm5iZiI6MTcxMjc2NDY2MiwicGF0aCI6Ii82ODM2MTE3NC8zMjExNzUxMDMtYzYwZDc0OWQtODBhZC00YmEwLTg3NjMtYzI2ZWE4N2M3ZTE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDEwVDE1NTc0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3MmU2NGJkYmQ5NTM5NmYwOTYwMmUzNTYyNWY0MzU4Zjc4YmJjZmNhYzQ1NDgwMmRmMTYwNjdiZDZkYjgxODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.doryIF-QQ1XPrev1-pAXP1R7KDSq9A43pkV6zoC0I8k)

Hi,
I'm sorry for any confusion caused by the previous explanation. In the scGPT cell type annotation tutorial, it requires two inputs for the ms dataset. While in our code, we require three inputs for the ms dataset is to ensure the stability of the reproduction results. We randomly divided the validation set into 10% and saved it as a separate h5ad file. Within adata.obs, the key "Factor Value [inferred cell type - author labels]" is entirely equivalent to the key "celltype." We had previously renamed it to unify the cell type label key, with no other alterations made. Regarding adata.obs["batch_id"], its purpose is to act as an identifier for distinguishing between the train set, validation set, and test set. By assigning a unique batch_id to each set, subsequent splitting becomes easier, allowing for clear identification of the different sets. We are currently engaged in a more substantial project and will gradually unveil more detailed codes. Stay tuned for updates! Let me know if you need additional details.
Best
